src/contracts/collections/positions.rs

```diff
@@ -46,18 +47,28 @@ impl Positions {
         account_id: AccountId,
         index: u32,
     ) -> Result<Position, ContractErrors> {
-        let (mut positions_length, mut positions) = self.get_value(account_id);
-
-        if index < positions_length {
-            let position = *positions.get(index as usize).unwrap();
-            positions_length -= 1;
-            positions.remove(index as usize);
-            self.positions
-                .insert(account_id, &(positions_length, positions));
-            Ok(position)
+        let positions_length = self.get_length(account_id);
+
+        if index >= positions_length {
+            return Err(ContractErrors::PositionNotFound);
+        }
+
+        let position = self.positions.get((account_id, index)).unwrap_or_default();
```

Comment:
It would be better to use `unwrap`. Although it has the same effect now, unwrap offers higher fault tolerance

Reply:
or even return `PositionNotFound`